### PR TITLE
Use DTO object in workflow pages 

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java [hunk 1]
```diff
@@ -851,4 +857,22 @@ private void resetGlobalProgress() {
         indexedFilter = 0;
     }
 
+    private void enableSortingByTextField(String type, String field) {
+        try {
+            indexRestClient.initiateClient();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java]: Must the index client initialised every time when this method is called?

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 2]
```diff
@@ -569,6 +560,91 @@ public void refresh(Process process) {
...
+    private QueryBuilder getQueryProjectArchived(boolean archived) throws DataException {
+        List<JSONObject> processesAccordingToArchive = serviceManager.getProjectService().findByArchived(archived);
+        BoolQueryBuilder query = new BoolQueryBuilder();
+        for (JSONObject singleProcess : processesAccordingToArchive) {
+            query.should(createSimpleQuery("project", getIdFromJSONObject(singleProcess), true));
+        }
+        return query;
+    }
+
+    public ProcessDTO getById(Integer id) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: New public method without JavaDoc. Please add the missing JavaDoc!

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 3]
```diff
@@ -569,6 +560,91 @@ public void refresh(Process process) {
...
+        return convertJSONObjectToDTO(findById(id), false);
+    }
+
+    public ProcessDTO getById(Integer id, boolean related) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: New public method without JavaDoc. Please add the missing JavaDoc!

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 4]
```diff
@@ -569,6 +560,91 @@ public void refresh(Process process) {
...
+        return convertJSONObjectToDTO(findById(id), related);
+    }
+
+    public Process convertDtoToBean(ProcessDTO processDTO) throws DAOException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: New public method without JavaDoc. Please add the missing JavaDoc!

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 5]
```diff
@@ -569,6 +560,91 @@ public void refresh(Process process) {
...
+        return serviceManager.getProcessService().find(processDTO.getId());
+    }
+
+    public ArrayList<Process> convertDtosToBeans(List<ProcessDTO> dtos) throws DAOException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: New public method without JavaDoc. Please add the missing JavaDoc!

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 6]
```diff
@@ -996,8 +1100,8 @@ public String getCreationDateAsString(Process process) {
      *            object
      * @return string
      */
-    public String getProgress(Process process) {
-        HashMap<String, Integer> tasks = calculationForProgress(process);
+    public String getProgress(Process process, ProcessDTO processDTO) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: You changed the method parameters without adjusting the JavaDoc!

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 7]
```diff
@@ -1211,6 +1361,22 @@ public boolean getContainsUnreachableSteps(Process process) {
     }
 
     /**
+     * Check whether the operation contains steps that are not assigned to a
+     * user or user group.
+     */
+    public boolean getContainsUnreachableSteps(ProcessDTO process) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc to method parameter and return value.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 8]
```diff
@@ -1224,6 +1390,19 @@ public boolean isImageFolderInUse(Process process) {
     }
 
     /**
+     * Check if there is one task in edit mode, where the user has the rights to
+     * write to image folder.
+     */
+    public boolean isImageFolderInUse(ProcessDTO process) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc to method parameter and return value.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 9]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
+
     /**
-     * Gets all not archvived Processes
+     * Gets all not archived Processes
      *
      * @return The list of processes.
      */
-    public List<Process> getNotArchivedProcesses() {
-        return processDAO.getNotArchivedProcesses();
+    public List<ProcessDTO> getNotArchivedProcesses(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 10]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * Get not closed processes.
      *
      * @return The list of processes.
      */
-    public List<Process> getNotClosedProcesses() {
-        return processDAO.getNotClosedProcesses();
+    public List<ProcessDTO> getNotClosedProcesses(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 11]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * Get not closed and not archived processes.
      *
      * @return The list of processes.
      */
-    public List<Process> getNotClosedAndNotArchivedProcesses() {
-        return processDAO.getNotClosedAndNotArchivedProcesses();
+    public List<ProcessDTO> getNotClosedAndNotArchivedProcesses(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 12]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * get not archived templates.
      *
      * @return The list of processes.
      */
-    public List<Process> getNotArchivedTemplates() {
-        return processDAO.getNotArchivedTemplates();
+    public List<ProcessDTO> getNotArchivedTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 13]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * Get all templates.
      *
      * @return The list of processes.
      */
-    public List<Process> getAllTemplates() {
-        return processDAO.getAllTemplates();
+    public List<ProcessDTO> getAllTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 14]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     /**
-     * get all processes, which are not a template
+     * Get all processes, which are not a template.
      *
-     * @return The list of processes.
+     * @return the list of processes.
      */
-    public List<Process> getAllWithoutTemplates() {
-        return processDAO.getAllWithoutTemplates();
+    public List<ProcessDTO> getAllWithoutTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 15]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     /**
-     * get all not archived processes which are not a template.
+     * Get all not archived processes which are not a template.
      *
-     * @return The list of processes.
+     * @return the list of processes.
      */
-    public List<Process> getAllNotArchivedWithoutTemplates() {
-        return processDAO.getAllNotArchivedWithoutTemplates();
+    public List<ProcessDTO> getAllNotArchivedWithoutTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 16]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * Get all not closed and not archived templates.
      *
      * @return The list of processes.
      */
-    public List<Process> getAllNotClosedAndNotArchivedTemplates() {
-        return processDAO.getAllNotClosedAndNotArchivedTemplates();
+    public List<ProcessDTO> getAllNotClosedAndNotArchivedTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 17]
```diff
@@ -2115,84 +2281,88 @@ private DocketData getDocketData(Process process) {
...
     }
 
     /**
      * Get all not closed templates.
      *
      * @return The list of processes.
      */
-    public List<Process> getAllNotClosedTemplates() {
-        return processDAO.getAllNotClosedTemplates();
+    public List<ProcessDTO> getAllNotClosedTemplates(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: Please adjust the JavaDoc for change method signature.

File: Kitodo/src/main/java/org/kitodo/services/data/ProjectService.java [hunk 18]
```diff
@@ -321,6 +321,14 @@ public JSONObject findByProcessId(Integer id) throws DataException {
         return findByUserId(getIdFromJSONObject(user));
     }
 
+    public ProjectDTO getById(Integer id) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProjectService.java]: Public method without JavaDoc. Please add missing JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/ProjectService.java [hunk 19]
```diff
@@ -321,6 +321,14 @@ public JSONObject findByProcessId(Integer id) throws DataException {
...
+        return convertJSONObjectToDTO(findById(id), false);
+    }
+
+    public ProjectDTO getById(Integer id, boolean related) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/ProjectService.java]: Public method without JavaDoc. Please add missing JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 20]
```diff
@@ -925,42 +1015,62 @@ public void executeDmsExport(Task step, boolean automatic) throws DataException,
     }
 
     /**
-     * get open tasks for current user.
+     * Get open tasks for current user.
      * 
      * @return The list of tasks.
      */
-    public List<Task> getOpenTasksForCurrentUser() {
+    public List<TaskDTO> getOpenTasksForCurrentUser(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Method signature is changed, please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 21]
```diff
@@ -925,42 +1015,62 @@ public void executeDmsExport(Task step, boolean automatic) throws DataException,
...
     }
 
     /**
-     * get open tasks without correction for current user.
+     * Get open tasks without correction for current user.
      * 
      * @return The list of tasks.
      */
-    public List<Task> getOpenTasksWithoutCorrectionForCurrentUser() {
+    public List<TaskDTO> getOpenTasksWithoutCorrectionForCurrentUser(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Method signature is changed, please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 22]
```diff
@@ -925,42 +1015,62 @@ public void executeDmsExport(Task step, boolean automatic) throws DataException,
...
     }
 
     /**
-     * get open not automatic tasks for current user.
+     * Get open not automatic tasks for current user.
      * 
      * @return The list of tasks.
      */
-    public List<Task> getOpenNotAutomaticTasksForCurrentUser() {
+    public List<TaskDTO> getOpenNotAutomaticTasksForCurrentUser(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Method signature is changed, please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/TaskService.java [hunk 23]
```diff
@@ -925,42 +1015,62 @@ public void executeDmsExport(Task step, boolean automatic) throws DataException,
...
 
     /**
-     * get open not automatix tasks without correction for current user
+     * Get open not automatic tasks without correction for current user.
      * 
-     * @return The list of tasks.
+     * @return the list of tasks.
      */
-    public List<Task> getOpenNotAutomaticTasksWithoutCorrectionForCurrentUser() {
+    public List<TaskDTO> getOpenNotAutomaticTasksWithoutCorrectionForCurrentUser(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/TaskService.java]: Method signature is changed, please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/base/SearchService.java [hunk 24]
```diff
@@ -312,6 +312,16 @@ public Long count(String query) throws DataException {
     }
 
     /**
+     * Display all documents for exact type.
+     *
+     * @return list of all documents
+     */
+    public List<JSONObject> findAllDocuments(String sort) throws DataException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/base/SearchService.java]: JavaDoc did not match to the method signature. Please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/data/base/SearchService.java [hunk 25]
```diff
@@ -662,6 +672,23 @@ protected Integer getIntegerPropertyForDTO(Object object, String key) {
     }
 
     /**
+     * Get size of related objects returned from ElasticSearch index.
+     *
+     * @param object
+     *            JSONObject as Object
+     * @return size of array with related objects
+     */
+    protected int getSizeOfRelatedPropertyForDTO(Object object, String key) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/data/base/SearchService.java]: JavaDoc did not match to the method signature. Please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 26]
```diff
@@ -641,6 +642,31 @@ public URI getProcessSubTypeURI(Process process, ProcessSubType processSubType,
...
+     * @param process
+     *            ProcessDTO object to get the sublocation for
+     * @param processSubType
+     *            The subType
+     * @param resourceName
+     *            the name of the single object (e.g. image) if null, the root
+     *            folder of the sublocation is returned
+     * @return The URI of the requested location
+     */
+    public URI getProcessSubTypeURI(ProcessDTO process, ProcessSubType processSubType, String resourceName) throws IOException {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: JavaDoc did not match to the method signature. Please adjust the JavaDoc.

File: Kitodo/src/main/java/org/kitodo/services/file/FileService.java [hunk 27]
```diff
@@ -660,6 +686,25 @@ public URI getProcessSubTypeURI(Process process, ProcessSubType processSubType,
...
+     *            FilenameFilter object
+     * @param process
+     *            ProcessDTO object
+     * @param processSubType
+     *            object
+     * @param resourceName
+     *            as String
+     * @return unmapped URI
+     */
+    public ArrayList<URI> getSubUrisForProcess(FilenameFilter filter, ProcessDTO process, ProcessSubType processSubType,
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/file/FileService.java]: JavaDoc did not match to the method signature. Please adjust the JavaDoc.

File: Kitodo/src/main/webapp/pages/inc_Prozessverwaltung/Prozesse_Liste.xhtml [hunk 28]
```diff
@@ -375,14 +375,13 @@
...
 
             <!-- Bearbeiten-Schaltknopf: Vorlagen -->
-            <h:commandLink action="#{ProzessverwaltungForm.editProcess}" id="action11"
-                           rendered="#{(LoginForm.maximaleBerechtigung == 1) and (ProzessverwaltungForm.modusAnzeige=='vorlagen')}"
-                           title="#{msgs.prozessBearbeiten}">
-                <h:graphicImage value="/pages/images/buttons/goInto.gif" alt="edit template"
+            <h:link outcome="/newpages/ProzessverwaltungBearbeiten"
+                    rendered="#{(LoginForm.maximaleBerechtigung == 1) and ProzessverwaltungForm.modusAnzeige=='vorlagen'}"
+                    title="#{msgs.prozessBearbeiten}">
+                <h:graphicImage value="/newpages/images/buttons/goInto.gif" alt="edit template"
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/pages/inc_Prozessverwaltung/Prozesse_Liste.xhtml]: Everything is moved from `newpages` to `pages` directory, so please did not change path to this image back to `newpages`.

File: Kitodo/src/test/java/org/kitodo/services/data/TemplateServiceIT.java [hunk 29]
```diff
@@ -173,8 +174,8 @@ public void shouldFindByProperty() throws Exception {
     public void shouldGetPropertiesSize() throws Exception {
         TemplateService templateService = new TemplateService();
 
-        Template template = templateService.find(1);
-        int actual = templateService.getPropertiesSize(template);
-        assertEquals("Template's properties size is not equal to given value!", 2, actual);
+        //TemplateDTO template = templateService.find(1);
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/services/data/TemplateServiceIT.java]: If this test is not needed anymore, then remove it or if this test is needed, fix it.

File: Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java (hunk 1 above)
Comment (Beacze) [inline Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java]: not sure, in general this thing must be adjusted and moved to the own button

File: Kitodo/src/main/webapp/pages/inc_Prozessverwaltung/Prozesse_Liste.xhtml (hunk 28 above)
Comment (Beacze) [inline Kitodo/src/main/webapp/pages/inc_Prozessverwaltung/Prozesse_Liste.xhtml]: rebase problem

**Comment by henning-gerhardt** [comment]
There is a merge conflict, please rebase / fix this conflict.

**Pull request merged**